src/core/Akka.Remote.TestKit/Player.cs

```diff
@@ -132,22 +131,26 @@ public void Enter(TimeSpan timeout, ImmutableList<string> names)
                 try
                 {
                     var askTimeout = barrierTimeout + Settings.QueryTimeout;
-                    //TODO: Wait?
+                    // Need to force barrier to wait here, so we can pass along a "fail barrier" message in the event
+                    // of a failed operation
                     _client.Ask(new ToServer<EnterBarrier>(new EnterBarrier(name, barrierTimeout)), askTimeout).Wait();
```

Comment:
The wait is correct - have to wait for the barrier to ACK before we can proceed. Has a timeout though, so it'll only wait up to 3-5 seconds IIRC.